### PR TITLE
WIP: New Group Chats

### PR DESCRIPTION
**Pull request by ingvar1995**
Hi all!

This is PR with working version of NGC based on toktok 0.2.3 release. For those who don't know - NGC are more powerful and usable version of group chats. NGC provide a lot of new features (features list is here: [link](https://github.com/ingvar1995/toxcore/blob/ngc_rebase/docs/DHT-Group-Chats.md)). NGC development, client and bots development took a lot of time for me and as longer I'm really disappointed with project development state and goals and don't see results of my effort, I decided to leave Tox development. This my farewell present to you. Feel free to continue my work!

You can test NGC using my client [Toxygen](https://github.com/toxygen-project/toxygen/tree/next_gen), 64-bit Linux users can [download prebuilt binary](https://dropmefiles.com/uN2QS). Add bot (909721BC8B933D9463D8FE32C604305D31AAB2C9AEF2406903F2ED4511319837E758679CE612) or join public gc (3AE831238B747BC32BD7D181A1F072AA3B4DA0A859781EDBFA62A70E5B9F66F4).

**UPD**: users of 64-bit Windows/Linux can use prebuild binary: [link](https://github.com/toxygen-project/toxygen/releases/tag/v0.5.0-prealpha)

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1162)

### CODE REVIEW

**Comment by iphydf** [comment]
@ingvar1995 can you help doing `git merge 8739f7fccb7cafc54ca0f5fa074c9a740f7048ba`?

File: toxcore/Messenger.c [hunk 1]
```diff
@@ -2820,8 +2820,12 @@ void do_messenger(Messenger *m, void *userdata)
     do_gc(m->group_handler, userdata);
     do_gca(m->group_announce);
     do_friends(m, userdata);
+<<<<<<< HEAD
```
Comment (iphydf) [inline toxcore/Messenger.c]: This is a merge marker.

**Comment by iphydf** [comment]
@ingvar1995 can you do 819fe534ea62b07340b20f18841d1ed93cf5caff?

**Comment by iphydf** [comment]
Never mind the merging stuff.

Can you check the diff between https://github.com/TokTok/c-toxcore/pull/1162 and your branch and, if you're confident that it doesn't break things, reset your branch to mine? This is the first step to being able to actually make all the commits in your branch work. I'm getting rid of the merge commits here.

**Comment by iphydf** [comment]
I also got rid of the PGC and its reversal.

**Comment by kpp** [comment]
@ingvar1995 What will happen if all members leave the chat? If the founder leaves the chat? If all members are offline?

**Comment by iphydf** [comment]
I've pushed the current ngc_rebase branch to [iphydf/ngc_rebase_backup](https://github.com/iphydf/c-toxcore/tree/ngc_rebase_backup). I will now force push my rebased version of this branch to this PR and close my rebased PR.

**Comment by iphydf** [comment]
Ok, it turns out I can't do that because ingvar1995/toxcore is a fork of irungentoo, not toktok, so I don't have permission to push to that branch. Closing this instead, and going to continue work in #1170.

**Pull request closed without merging**